internal/resource/config/oauth/client/oauth_client_resource.go

```diff
@@ -0,0 +1,1579 @@
+			resp.Diagnostics.AddError("persistent_grant_expiration_type must be configured to \"OVERRIDE_SERVER_DEFAULT\" to modify the other persistent_grant_expiration values.", "")
+		}
+	}
+	if (internaltypes.IsDefined(model.PersistentGrantExpirationTime) || internaltypes.IsDefined(model.PersistentGrantExpirationTimeUnit)) && model.PersistentGrantExpirationType.ValueString() != "OVERRIDE_SERVER_DEFAULT" {
+		resp.Diagnostics.AddError("persistent_grant_expiration_type must be configured to \"OVERRIDE_SERVER_DEFAULT\" to modify the other persistent_grant_expiration values.", "")
+	}
+
+	// Refresh Token Rolling Validation
+	if model.RefreshTokenRollingIntervalType.ValueString() == "OVERRIDE_SERVER_DEFAULT" && !internaltypes.IsDefined(model.RefreshTokenRollingInterval) {
+		resp.Diagnostics.AddError("refresh_token_rolling_interval must be configured when refresh_token_rolling_interval_type is \"OVERRIDE_SERVER_DEFAULT\".", "")
+	}
+	if internaltypes.IsDefined(model.RefreshTokenRollingInterval) && model.RefreshTokenRollingIntervalType.ValueString() != "OVERRIDE_SERVER_DEFAULT" {
+		resp.Diagnostics.AddError("refresh_token_rolling_interval_type must be configured to \"OVERRIDE_SERVER_DEFAULT\" to modify the refresh_token_rolling value.", "")
+	}
+
+	//  Client Auth Defined
+	var clientAuthDefined bool
+	var clientAuthAttributes map[string]attr.Value
+	if internaltypes.IsDefined(model.ClientAuth) {
+		clientAuthDefined = true
+		clientAuthAttributes = model.ClientAuth.Attributes()
+		if internaltypes.IsDefined(clientAuthAttributes["type"]) {
+			clientAuthType := clientAuthAttributes["type"].(types.String).ValueString()
+			switch clientAuthType {
+			case "PRIVATE_KEY_JWT":
+				if !internaltypes.IsNonEmptyObj(model.JwksSettings) {
+					resp.Diagnostics.AddError("jwks_settings must be defined when client_auth is configured to \"PRIVATE_KEY_JWT\".", "")
+				}
+			case "CERTIFICATE":
+				if !internaltypes.IsDefined(clientAuthAttributes["client_cert_subject_dn"]) || !internaltypes.IsDefined(clientAuthAttributes["client_cert_issuer_dn"]) {
+					resp.Diagnostics.AddError("client_cert_subject_dn and client_cert_issuer_dn must be defined when client_auth is configured to \"CERTIFICATE\".", "")
+				}
+			}
+		}
+	} else {
+		clientAuthDefined = false
+	}
+
+	// Grant Types Validation
+	// grant_types is required, don't need nil check here
+	var hasCibaGrantType bool
+	for _, grantType := range model.GrantTypes.Elements() {
+		grantTypeVal := grantType.(types.String).ValueString()
+		if grantTypeVal == "CLIENT_CREDENTIALS" {
+			if clientAuthDefined {
+				clientAuthType := clientAuthAttributes["type"].(types.String).ValueString()
+				clientAuthSecret := clientAuthAttributes["secret"].(types.String).ValueString()
+				if clientAuthType != "NONE" && clientAuthSecret == "" {
+					resp.Diagnostics.AddError("client_auth must be defined when \"CLIENT_CREDENTIALS\" is included in grant_types.", "")
```

Comment:
This error message doesn't seem right

Reply:
The default value is for the type equal to NONE, and the secret is empty. So if it's not different, this would mean it's not set properly.

Reply:
I was confused on this one because in this if statement `client_auth` is defined, and the error message is implying that it isn't

Reply:
ahh, I see. I'll update this description.